File: split/overbar.pas
function overbar(b: halfword; k, t: scaled): halfword;
var
  p, q: halfword;
begin
  p := newkern(k);
  mem[p].hh.rh := b;
  q := fractionrule(t);
  mem[q].hh.rh := p;
  p := newkern(t);
  mem[p].hh.rh := q;
  overbar := vpackage(p, 0, 1, 1073741823);
end;

File: cpp/overbar.cpp
#include "overbar.h"
#include "newkern.h"
#include "fractionrule.h"
#include "vpackage.h"

halfword overbar(halfword b, scaled k, scaled  t)
{
	auto p = newkern(k);
	link(p) = b;
	auto q = fractionrule(t);
	link(q) = p;
	p = newkern(t);
	link(p) = q;
	return vpackage(p, 0, 1, max_dimen);
}
